### PR TITLE
Added new indev type LV_INDEV_TYPE_NAVIGATION

### PR DESCRIPTION
**Pull request by fhorinek**
input type usable with small keypad
LV_KEY_LEFT and LV_KEY_RIGHT keys are used for navigation and value edit
LV_KEY_ENTER short click enter and leaves edit mode if focused object is editable
(optional) LV_KEY_ESC exit edit mode

### CODE REVIEW

**Comment by kisvegabor** [comment]
Hi, 

I've tried this new indev type and got a couple of questions about the expected behavior. I think it should work like this:
- Non-editable objects (e.g. button)
  - ENTER: simply send PRESS, RELEASE, LONG_PRESS, LONG_PRESS_REPEAT signals and events. LONG_PRESS_REPEAT should work as it is handled with the keypad type.
   - RIGHT: Focus the next objects
   - LEFT: Focus the previous object
- Editable objects (e.g. slider):
  - In navigate mode
     - ENTER: On press send go to edit mode. Other events are ignored
     - RIGHT: Focus the next objects
     - LEFT: Focus the previous object
  - In editing mode
     - ENTER: Handle press and release, but on long-press leave edit mode 
     - RIGHT/LEFT: On press send the key, after long press repeat it (same as with non-editable objects)

Basically I've found only two differences in the current implementation:
1. Edit mode should be left on enter log press, to allow sending a click event.
2. Repeating should use `i->driver.long_press_rep_time`.

What do you think?

**Comment by fhorinek** [comment]
Hi,

this type of navigation meant for the device with large number of configuration options with multiple levels. Similar to android settings menu.

The long press to exit the edit mode seams a bit confusing since it is not the same process as to enter the edit mode. When I handed the device to a few colleagues, no one was able to exit the edit mode without asking. Single click is just faster and more intuitive.

Also I do not need to send single click to editable widget, but the long press might be useful as it might be a way to restore the default value. This behavior is also not intuitive, but it might be useful for more advanced users.

RE: long_press_rep_time, will change

**Comment by kisvegabor** [comment]
> Also I do not need to send single click to editable widget, but the long press might be useful as it might be a way to restore the default value. This behavior is also not intuitive, but it might be useful for more advanced users.

It's mainly an issue with the button matrix and keyboard. If you navigate to a button (or letter) and hit enter to press that key, the key will be sent, and navigate mode will be returned hence no other key can be selected. 
It's also a problem with spinbox, where enter click is used to go to the next digit. 

I understand your concern but we can't leave widgets unusable. Do you have any idea to solve it?

**Comment by fhorinek** [comment]
Is there any way how to get object type?

**Comment by fhorinek** [comment]
I can add exceptions for the problematic widgets to use long press for keyboards, button matrix and spinbox...

**Comment by kisvegabor** [comment]
I need to mention that some widget leave edit mode on release although encoder would leave it on long press. For example, the `slider` can do anything on enter release, so it leaves edit mode. But `list`, `btnmartix`, `keyboard`, `spinbox`, `textarea` leave it on on long press.

You can try by creating an encoder input device from your buttons.

In light of that, is it still not an option for you?

**Comment by fhorinek** [comment]
Actually I did not notice that behavior in encoder. I do understand that the exceptions for yet another input method trough out the whole widget code might be unnecessary clutter.

But still I want to have cleaner solution. Can I modify the encoder proc to accept arrow keys and generate the enc_diff value internally and pass other keys?

**Comment by kisvegabor** [comment]
> But still I want to have cleaner solution. Can I modify the encoder proc to accept arrow keys and generate the enc_diff value internally and pass other keys?

That sounds great!
